test/parallel/test-dotenv-edge-cases.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('node:assert');
+const { describe, it } = require('node:test');
+
+const validEnvFilePath = '../fixtures/dotenv/valid.env';
+const relativePath = '../fixtures/dotenv/node-options.env';
+
+describe('.env supports edge cases', () => {
+
+  it('should use the last --env-file declaration', async () => {
```

Comment:
```suggestion
  // TODO(@anonrig): This might change in the future, for example to support loading all files
  it('should use the last --env-file declaration', async () => {
```

Reply:
@GeoffreyBooth Would you mind keeping the pull request as it is, since we have them listed in the issue: https://github.com/nodejs/node/issues/49148? I'd like to land this and I'm happy to follow up on a different pull request as soon as this PR lands. (Going through the review + CI process is exhausting)

Reply:
These are just suggestions, I’ve already approved, so feel free to disregard if you want.